flink-state-backends/flink-statebackend-rocksdb/src/main/java/org/apache/flink/contrib/streaming/state/restore/RocksDBIncrementalRestoreOperation.java

```diff
@@ -414,90 +424,70 @@ private void restoreWithRescaling(Collection<KeyedStateHandle> restoreStateHandl
+                        restoreTempDBInstanceFromLocalState(stateHandle)) {
 
-                List<ColumnFamilyHandle> tmpColumnFamilyHandles =
-                        tmpRestoreDBInfo.columnFamilyHandles;
+                    List<ColumnFamilyHandle> tmpColumnFamilyHandles =
+                            tmpRestoreDBInfo.columnFamilyHandles;
 
-                // Clip all tmp db to Range [startKeyGroupPrefixBytes, stopKeyGroupPrefixBytes)
-                RocksDBIncrementalCheckpointUtils.clipColumnFamilies(
-                        tmpRestoreDBInfo.db,
-                        tmpColumnFamilyHandles,
-                        startKeyGroupPrefixBytes,
-                        stopKeyGroupPrefixBytes);
-
-                // Export all the Column Families
-                Map<RegisteredStateMetaInfoBase, ExportImportFilesMetaData> exportedCFAndMetaData =
-                        RocksDBIncrementalCheckpointUtils.exportColumnFamilies(
-                                tmpRestoreDBInfo.db,
-                                tmpColumnFamilyHandles,
-                                tmpRestoreDBInfo.stateMetaInfoSnapshots,
-                                exportCfBasePath);
-
-                exportedCFAndMetaData.forEach(
-                        (stateMeta, cfMetaData) -> {
-                            if (!cfMetaData.files().isEmpty()) {
-                                cfMetaDataToImport.putIfAbsent(stateMeta, new ArrayList<>());
-                                cfMetaDataToImport.get(stateMeta).add(cfMetaData);
-                            }
-                        });
-            } finally {
-                cleanUpPathQuietly(downloadRequest.getDownloadDestination());
+                    // Clip all tmp db to Range [startKeyGroupPrefixBytes, stopKeyGroupPrefixBytes)
+                    RocksDBIncrementalCheckpointUtils.clipColumnFamilies(
+                            tmpRestoreDBInfo.db,
+                            tmpColumnFamilyHandles,
+                            startKeyGroupPrefixBytes,
+                            stopKeyGroupPrefixBytes);
+
+                    // Export all the Column Families
+                    List<Pair<RegisteredStateMetaInfoBase, ExportImportFilesMetaData>>
+                            exportedCFAndMetaData =
+                                    RocksDBIncrementalCheckpointUtils.exportColumnFamilies(
+                                            tmpRestoreDBInfo.db,
+                                            tmpColumnFamilyHandles,
+                                            tmpRestoreDBInfo.stateMetaInfoSnapshots,
+                                            exportCfBasePath);
+
+                    for (Pair<RegisteredStateMetaInfoBase, ExportImportFilesMetaData> entry :
+                            exportedCFAndMetaData) {
+                        ExportImportFilesMetaData cfMetaData = entry.getValue();
+                        // TODO: method files() doesn't exist in the RocksDB API
```

Comment:
@mayuehappy Is it correct to remove this code? I could not find a files() method in the RocksDB API.